src/index.js

```diff
@@ -54,226 +56,301 @@ function buildNametoIdMap(objectArr, type) {
-      } else {
-        // create a new library
-        res = await createLibrary(
-          lib.name,
-          lib.description,
-          code,
-          lib.language
-        );
-        core.info(`created library: ${lib.name}`);
-      }
-      libraryDict[res.data.versionId] = { ...lib, id: res.data.id };
+// Create or update a transformation.
+async function upsertTransformations(transformations, transformationDict) {
+  core.info(`Upserting transformations`);
+  for (let i = 0; i < transformations.length; i++) {
+    let tr = transformations[i];
+    let code = fs.readFileSync(tr.file, "utf-8");
+    let res;
+    if (transformationNameToId[tr.name]) {
+      // update existing transformer and get a new versionId
+      let id = transformationNameToId[tr.name];
+      res = await updateTransformation(
+        id,
+        tr.name,
+        tr.description,
+        code,
+        tr.language
+      );
+    } else {
+      // create new transformer
+      res = await createTransformation(
+        tr.name,
+        tr.description,
+        code,
+        tr.language
+      );
     }
-    core.info("Libraries create/update done!");
-
-    let transformationTest = [];
-    let librariesTest = [];
-
-    core.info("Building test suite...");
-    for (let i = 0; i < Object.keys(transformationDict).length; i++) {
-      let trVersionId = Object.keys(transformationDict)[i];
-      let testInputPath =
-        transformationDict[trVersionId]["test-input-file"] || "";
-      let testInput = testInputPath
-        ? JSON.parse(fs.readFileSync(testInputPath))
-        : "";
-      if (testInput) {
-        transformationTest.push({ versionId: trVersionId, testInput });
-      } else {
-        core.info(
-          `No test input provided. Testing ${transformationDict[trVersionId].name} with default payload`
-        );
-        transformationTest.push({ versionId: trVersionId });
-      }
+    transformationDict[res.data.versionId] = { ...tr, id: res.data.id };
+  }
+}
+
+// Create or update a library.
+async function upsertLibraries(libraries, libraryDict) {
+  core.info(`Upserting libraries`);
+  for (let i = 0; i < libraries.length; i++) {
+    let lib = libraries[i];
+    let code = fs.readFileSync(lib.file, "utf-8");
+    let res;
+    if (libraryNameToId[lib.name]) {
+      // update library and get a new versionId
+      let id = libraryNameToId[lib.name];
+      res = await updateLibrary(id, lib.description, code, lib.language);
+      core.info(`Updated library: ${lib.name}`);
+    } else {
+      // create a new library
+      res = await createLibrary(lib.name, lib.description, code, lib.language);
+      core.info(`Created library: ${lib.name}`);
     }
+    libraryDict[res.data.versionId] = { ...lib, id: res.data.id };
+  }
+}
 
-    for (let i = 0; i < Object.keys(libraryDict).length; i++) {
-      librariesTest.push({ versionId: Object.keys(libraryDict)[i] });
+// Build the test suite.
+async function buildTestSuite(
```

Comment:
Make sure we are following the approach of using params that are absolutely necessary and return librariesTest, transformationsTest from this function ?